src/psyclone/domain/gocean/transformations/gocean_move_iteration_boundaries_inside_kernel_trans.py

```diff
@@ -0,0 +1,226 @@
+        :type node: :py:class:`psyclone.psyGen.CodedKern`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if the node is not a CodedKern.
+         '''
+
+        if not isinstance(node, CodedKern):
+            raise TransformationError("Error in {0} transformation. "
+                                      "This transformation can only be applied"
+                                      " to CodedKern nodes.".format(self.name))
+
+    def apply(self, node, options=None):
+        '''Apply this transformation to the supplied node.
+
+        :param node: the node to transform.
+        :type node: :py:class:`psyclone.psyGen.CodedKern`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :returns: 2-tuple of new schedule and memento of transform.
+        :rtype: (:py:class:`psyclone.gocean1p0.GOInvokeSchedule`, \
+                 :py:class:`psyclone.undoredo.Memento`)
+
+        '''
+        self.validate(node)
+
+        # Get useful references
+        invoke_st = node.ancestor(InvokeSchedule).symbol_table
+        inner_loop = node.parent.parent
+        outer_loop = inner_loop.parent.parent
+        cursor = outer_loop.position
+
+        # Create new symbols in the PSylayer and initialise them with
+        inv_xstart = invoke_st.symbol_from_tag(
+            "xstart_" + node.name, root_name="xstart", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+        inv_xstop = invoke_st.symbol_from_tag(
+            "xstop_" + node.name, root_name="xstop", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+        inv_ystart = invoke_st.symbol_from_tag(
+            "ystart_" + node.name, root_name="ystart", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+        inv_ystop = invoke_st.symbol_from_tag(
+            "ystop_" + node.name, root_name="ystop", symbol_type=DataSymbol,
+            datatype=INTEGER_TYPE)
+
+        assign1 = Assignment.create(Reference(inv_xstart),
+                                    inner_loop.lower_bound())
+        outer_loop.parent.children.insert(cursor, assign1)
+        cursor = cursor + 1
+        assign2 = Assignment.create(Reference(inv_xstop),
+                                    inner_loop.upper_bound())
+        outer_loop.parent.children.insert(cursor, assign2)
+        cursor = cursor + 1
+        assign3 = Assignment.create(Reference(inv_ystart),
+                                    outer_loop.lower_bound())
+        outer_loop.parent.children.insert(cursor, assign3)
+        cursor = cursor + 1
+        assign4 = Assignment.create(Reference(inv_ystop),
+                                    outer_loop.upper_bound())
+        outer_loop.parent.children.insert(cursor, assign4)
+
+        # Update Kernel Call (only works with the gen_ocl_code)
```

Comment:
I was being lazy and only working with my use case, I now update the Arguments.args instead of the raw_args_list() and it should work in every scenario. I added a new test showing that it also work as expected with the Psy-layer produced by the PSyIR Fortran Writer :)